### PR TITLE
Add support for file spec requirements

### PR DESCRIPTION
**Pull request by tumido**
## Related Issues and Dependencies

Fixes https://github.com/thoth-station/micropipenv/issues/99

## This introduces a breaking change

- [ ] Yes
- [x] No

## This Pull Request implements

Add `file` attribute support for package resolution

### CODE REVIEW

**Comment by tumido** [comment]
Now also with tests :slightly_smiling_face:

**Comment by tumido** [comment]
eh.. it could have been moved around a bit and included into [the `if/else` above](https://github.com/thoth-station/micropipenv/pull/100/files#diff-73fbdb2339888190b1cf072e5dc66c32R885-R887) What do you think? Would that be better?

File: tests/data/install/pipenv_file/setup [hunk 1]
```diff
@@ -0,0 +1,15 @@
+"""A setup.py file for requirements.txt in editable mode."""
```
Comment (fridex) [inline tests/data/install/pipenv_file/setup:1]: Is this file needed? It feels like only Pipfile.lock is required.

File: tests/test_micropipenv.py [hunk 2]
```diff
@@ -126,6 +126,13 @@ def test_install_pipenv_vcs(venv):
         subprocess.run(cmd, check=True, env={"MICROPIPENV_PIP_BIN": get_pip_path(venv), "MICROPIPENV_DEBUG": "1"})
         assert str(venv.get_version("daiquiri")) == "2.0.0"
 
+def test_install_pipenv_file(venv):
+    """Test invoking installation using information in Pipfile.lock, a file mode is used."""
+    cmd = [os.path.join(venv.path, "bin", "python3"), micropipenv.__file__, "install", "--method", "pipenv"]
+    with cwd(os.path.join(_DATA_DIR, "install", "pipenv_file")):
+        subprocess.run(cmd, check=True, env={"MICROPIPENV_PIP_BIN": get_pip_path(venv), "MICROPIPENV_DEBUG": "1"})
+        assert str(venv.get_version("daiquiri")) == "2.0.0"
```
Comment (fridex) [inline tests/test_micropipenv.py:134]: You can also assert on the second package installed to check that the installation respects Pipfile.lock:

```python
assert  str(venv.get_version("python-json-logger")) == "0.1.11"
```

File: tests/data/install/pipenv_file/setup (hunk 1 above)
Comment (tumido) [reply tests/data/install/pipenv_file/setup:1]: It's not needed. I've used `tests/data/install/pipenv_editable` folder as a bootstrap for my test case fixtures and didn't think twice about it. Removing it now.

**Comment by tumido** [comment]
@fridex is the

> To be fully compatible, we could include direct reference support later on. https://www.python.org/dev/peps/pep-0440/#direct-references

something you'd like to see as a part of this PR or was that just a general comment? :slightly_smiling_face:

**Comment by fridex** [comment]
> @fridex is the
> 
> > To be fully compatible, we could include direct reference support later on. https://www.python.org/dev/peps/pep-0440/#direct-references
> 
> something you'd like to see as a part of this PR or was that just a general comment?

Just a general comment :+1:

**Comment by tumido** [comment]
Hey @fridex can we please get a micropipenv release with this change? :slightly_smiling_face:

We kinda desperately need it for https://github.com/aicoe-aiops/openshift-sme-mailing-list-analysis . Btw, once released, will the https://github.com/thoth-station/s2i-minimal-notebook and https://github.com/AICoE/s2i-custom-notebook based s2i builds pick up this new release automatically or does it have to be updated anywhere else? (Maybe @harshad16 knows this better) I think the Dockerfiles in https://github.com/thoth-station/s2i-thoth would have to be updated as well, right?

**Comment by fridex** [comment]
> Hey @fridex can we please get a micropipenv release with this change? 


Definitely, we can release it. :+1: 

> We kinda desperately need it for https://github.com/aicoe-aiops/openshift-sme-mailing-list-analysis . Btw, once released, will the https://github.com/thoth-station/s2i-minimal-notebook and https://github.com/AICoE/s2i-custom-notebook based s2i builds pick up this new release automatically or does it have to be updated anywhere else? (Maybe @harshad16 knows this better) I think the Dockerfiles in https://github.com/thoth-station/s2i-thoth would have to be updated as well, right?

The Dockerfile(s) need to be updated manually (as of now).
